frontend/pulse3d/src/components/uploadForm/AnalysisParamForm.js

```diff
@@ -143,65 +171,125 @@ export default function AnalysisParamForm({
       // need to validate start and end time together
       validateWindowBounds(updatedParams);
     }
-    if (newParams.prominenceFactor !== undefined) {
-      validateProminenceFactor(updatedParams);
+    if (newParams.prominenceFactorPeaks !== undefined) {
+      validateProminenceFactorMin(updatedParams);
+    }
+    if (newParams.prominenceFactorValleys !== undefined) {
+      validateProminenceFactorMax(updatedParams);
     }
-    if (newParams.widthFactor !== undefined) {
-      validateWidthFactor(updatedParams)
+    if (newParams.widthFactorPeaks !== undefined) {
+      validateWidthFactorMin(updatedParams)
+    }
+    if (newParams.widthFactorValleys !== undefined) {
+      validateWidthFactorMax(updatedParams)
```

Comment:
not necessarily a change, but you can just do `if (newParams.... )` or `if (!newParams...)`, you don't have to compare to undefined because undefined is a falsey value